python/triton/runtime/jit.py

```diff
@@ -591,7 +591,9 @@ def run(self, *args, grid, warmup, **kwargs):
 
         # Execute pre run hooks with args and kwargs
         for hook in self.pre_run_hooks:
-            hook(*args, **kwargs)
+            result = hook(self, *args, grid=grid, warmup=warmup, **kwargs)
```

Comment:
You probably want to document why returning from `if result` to specify the behavior of a hook function.

Reply:
Can do

Reply:
Maybe the return value should be named as `early_return` instead of `result`? My two cents